lgc/builder/DescBuilder.cpp

```diff
@@ -184,9 +184,22 @@ Value *DescBuilder::CreateGetDescPtr(ResourceNodeType descType, unsigned descSet
     Module *module = GetInsertPoint()->getModule();
     descPtr = module->getGlobalVariable(globalName, /*AllowInternal=*/true);
     if (!descPtr) {
-      descPtr = new GlobalVariable(*module, node->immutableValue->getType(),
-                                   /*isConstant=*/true, GlobalValue::InternalLinkage, node->immutableValue, globalName,
-                                   nullptr, GlobalValue::NotThreadLocal, ADDR_SPACE_CONST);
+      // We need to create the global variable as it does not already exist.
+      // First construct the immutable value as an array of <4 x i32>, for use as the initialzier of the
+      // global variable.
+      SmallVector<Constant *, 4> immutableArray;
```

Comment:
Doesn't really matter but if you know you'll never put more than 4 things in this vector then `std::array<Constant *, 4>` is a bit lighter weight.

Reply:
This is arbitrary size. It is the "dwords" array a couple of lines down that is always 4.

Reply:
Oh OK. But then you can drop the ", 4", unless you have special knowledge that 4 is the most optimalest value.

Reply:
I didn't know you could do that until I saw someone else do it the other day.